sklearn/model_selection/_split.py

```diff
@@ -620,7 +620,10 @@ def __init__(self, n_splits='warn', shuffle=False, random_state=None):
         super().__init__(n_splits, shuffle, random_state)
 
     def _make_test_folds(self, X, y=None):
-        rng = self.random_state
+        if self.shuffle:
```

Comment:
Do we really need to check shuffle here?

Otherwise I agree this would require some non-regression test, possibly for all CV iterators